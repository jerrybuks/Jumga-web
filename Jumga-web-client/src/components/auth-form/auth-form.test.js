import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import AuthForm from './index';

describe('AuthForm Component', () => {
	let wrapper;
	let shallow;
	let mocklocation, mockEmailSignInStart, mockGoogleSignInStart, mockSignUpStart;

	beforeEach(() => {
		mocklocation = {
			state: 'sign up'
		};
		mockEmailSignInStart = jest.fn();
		mockGoogleSignInStart = jest.fn();
		mockSignUpStart = jest.fn();

		const mockProps = {
			location: mocklocation,
			emailSignInStart: mockEmailSignInStart,
			googleSignInStart: mockGoogleSignInStart,
			signUpStart: mockSignUpStart,
			isLoggingIn: false
		};
		shallow = createShallow();
		wrapper = shallow(<AuthForm {...mockProps} />);
	});

	it('should render AuthForm component', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should call googleSignInStart when Sign up button is clicked', () => {
		wrapper.find('GoogleButton').simulate('click');
		expect(mockGoogleSignInStart).toHaveBeenCalled();
	});
});
