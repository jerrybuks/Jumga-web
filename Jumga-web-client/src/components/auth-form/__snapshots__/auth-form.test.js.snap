// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm Component should render AuthForm component 1`] = `ShallowWrapper {}`;
